crates/executor/src/blockchain_tree/mod.rs

```diff
@@ -480,13 +490,20 @@ impl<DB: Database, C: Consensus, EF: ExecutorFactory> BlockchainTree<DB, C, EF>
     ///
     /// Returns `Ok` if the blocks were canonicalized, or if the blocks were already canonical.
     pub fn make_canonical(&mut self, block_hash: &BlockHash) -> Result<(), Error> {
-        let chain_id = if let Some(chain_id) = self.block_indices.get_blocks_chain_id(block_hash) {
-            chain_id
-        } else {
-            // If block is already canonical don't return error.
-            if self.block_indices.is_block_hash_canonical(block_hash) {
-                return Ok(())
+        // If block is already canonical don't return error.
+        if self.block_indices.is_block_hash_canonical(block_hash) {
+            let td = self
+                .externals
+                .shareable_db()
+                .header_td(block_hash)?
```

Comment:
This is not correct, make canonical makes block from the tree be saved to database, so `header_td` will always return false here.